rustls/tests/api.rs

```diff
@@ -956,6 +958,86 @@ fn client_trims_terminating_dot() {
     }
 }
 
+#[test]
+fn server_ignores_sni_with_ip_address() {
+    fn insert_ip_address_server_name(msg: &mut Message) -> Altered {
+        if let MessagePayload::Handshake { parsed, encoded } = &mut msg.payload {
+            if let HandshakePayload::ClientHello(ch) = &mut parsed.payload {
+                for mut ext in ch.extensions.iter_mut() {
+                    if let ClientExtension::ServerName(snr) = &mut ext {
+                        snr.clear();
+                        snr.push(
+                            ServerNameExtensionItem::read_bytes(b"\x00\x00\x071.1.1.1").unwrap(),
+                        );
+                    }
+                }
+            }
+
+            *encoded = Payload::new(parsed.get_encoding());
+        }
+
+        Altered::InPlace
+    }
+
+    check_sni_error(
+        insert_ip_address_server_name,
+        Error::General("no server certificate chain resolved".to_string()),
+    );
+}
+
+#[test]
+fn server_rejects_sni_with_illegal_dns_name() {
+    fn insert_illegal_server_name(msg: &mut Message) -> Altered {
+        if let MessagePayload::Handshake { parsed, encoded } = &mut msg.payload {
```

Comment:
Same here.

(Also looks like there's quite a bit of duplication between the two tests? Not sure if that can all be moved into `check_sni_error()`, but a helper function would probably be nicer?)

Reply:
Have added a helper function, and then used it in the existing two other SNI editing tests.